libs/types-common/test/Test/Domain.hs

```diff
@@ -0,0 +1,27 @@
+module Test.Domain
+  ( tests,
+  )
+where
+
+import Data.Domain (DomainText (DomainText), domainText, mkDomain)
+import qualified Data.Text as Text
+import Imports
+import Test.Tasty
+import Test.Tasty.QuickCheck
+
+tests :: TestTree
+tests =
+  testGroup
+    "Domain"
+    [ testGroup "serialization" testDomainSerialization
+    ]
+
+testDomainSerialization :: [TestTree]
+testDomainSerialization =
+  [ testProperty "Arbitrary DomainText generates valid domains" $
```

Comment:
It's good to phrase it like this, and not "mkDomain works for *all* inputs".  :)

Reply:
You could run `mkDomain` on some invalid (and valid) domains given as string literals, if you really wanted to.

Reply:
I'll do that, similar to the tests for handles.

Reply:
added the tests, done.